webtau-http/src/main/java/com/twosigma/webtau/http/validation/HeaderDataNode.java

```diff
@@ -93,4 +99,12 @@ public String toString() {
     public DataNode statusCode() {
         return dataNode.get("statusCode");
     }
+
+    private String findMatchingCaseInsensitiveKey(String name) {
+        String lowerCaseName = name.toLowerCase();
+        return dataNode.asMap().keySet().stream()
+                .filter(k -> k.toLowerCase().equals(lowerCaseName))
+                .findFirst()
+                .orElse("");
```

Comment:
Why not return the optional from `findFirst` and then instead of `key.isEmpty()` use `optional.isPresent()`?

Reply:
old habits. let me change that.